### PR TITLE
AKS subnet nsg rules and default route table

### PR DESCRIPTION
**Pull request by rajeshkaremane**

### CODE REVIEW

File: infra/network/aks-subnet-nsg.bicep [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+param deploymentDate string = utcNow('yyyyMMdd-HHmmss')
+
+@description('Required. Array of Security Rules to deploy to the Network Security Group. When not provided, an NSG including only the built-in roles will be deployed.')
+param securityRules array = []
+
+var commonTags = {
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+  Purpose: 'ADP-NSG'
```
Comment (DaRoszko) [inline infra/network/aks-subnet-nsg.bicep:20]: Purpose: ADP-NSG - AKS Subnet

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: infra/network/aks-subnet-nsg.bicep (hunk 1 above)
Comment (DaRoszko) [reply infra/network/aks-subnet-nsg.bicep:20]: Actually - Shouldn't we have a generic NSG template that just takes a list of configuration so we can have many NSGs etc? @ppotturi  @ajaysoni-cap  @kbabigumira

File: .azuredevops/deploy-platform-core.yaml [hunk 2]
```diff
@@ -60,6 +60,9 @@ extends:
           - name: route-table
             path: infra/network
             resourceGroupName: $(virtualNetworkResourceGroup)
+          - name: aks-subnet-nsg
```
[the remark this answers is not in the document]
Comment (ajaysoni-cap) [reply .azuredevops/deploy-platform-core.yaml:63]: We agreed to keep it separate for now and then revisit in the future.

File: infra/network/aks-subnet-nsg.bicep (hunk 1 above)
Comment (rajeshkaremane) [reply infra/network/aks-subnet-nsg.bicep]: Updated the code according to our discussion this morning.

**Review by DaRoszko — APPROVED** [review]

**Pull request merged**